dango/bitcoin/src/execute.rs

```diff
@@ -0,0 +1,678 @@
+        if cumulative_signatures.len() >= cfg.multisig.threshold() as usize {
+            Some(OutboundConfirmed {
+                id,
+                transaction: OUTBOUNDS.load(ctx.storage, id)?,
+                signatures: cumulative_signatures,
+            })
+        } else {
+            None
+        },
+    )?)
+}
+
+fn transfer_remote(
+    ctx: MutableCtx,
+    req: TransferRemoteRequest,
+    amount: Uint128,
+) -> anyhow::Result<Response> {
+    // Ensure only the gateway can call this function, that is responsible also
+    // to deduct the withdrawal fees. The amount here is the net amount to withdraw.
+    ensure!(
+        ctx.sender == ctx.querier.query_gateway()?,
+        "only gateway can call `transfer_remote`"
+    );
+
+    let TransferRemoteRequest::Bitcoin { recipient } = req else {
+        bail!("incorrect TransferRemoteRequest type! expected: Bitcoin, found: {req:?}");
+    };
+
+    let cfg = CONFIG.load(ctx.storage)?;
+
+    // Ensure the recipient address is valid.
+    check_bitcoin_address(&recipient, cfg.network)?;
+
+    // TODO: remove this check?
+    ensure!(
+        recipient != cfg.vault,
+        "cannot withdraw to the vault address"
+    );
+
+    // If there is already a withdrawal to the same recipient, accumulate the amount.
+    OUTBOUND_QUEUE.may_update(ctx.storage, recipient, |outbound| -> StdResult<_> {
+        Ok(outbound.unwrap_or_default().checked_add(amount)?)
+    })?;
+
+    Ok(Response::new())
+}
+
+fn create_deposit_address(ctx: MutableCtx) -> anyhow::Result<Response> {
+    let (index, _) = ADDRESS_INDEX.increment(ctx.storage)?;
+
+    // Add a deposit address for the user if he hasn't already one.
+    ADDRESSES.may_update(
+        ctx.storage,
+        ctx.sender,
+        |maybe_address| match maybe_address {
+            Some(_) => bail!("you already have a deposit address"),
+            None => Ok(index),
+        },
+    )?;
+
+    Ok(Response::new())
+}
+
+#[cfg_attr(not(feature = "library"), grug::export)]
+pub fn cron_execute(ctx: SudoCtx) -> anyhow::Result<Response> {
+    let cfg = CONFIG.load(ctx.storage)?;
+
+    // Take the pending outbound transfers from the storage.
+    let outputs = OUTBOUND_QUEUE.drain(ctx.storage, None, None)?;
+
+    // If there's no pending outbound transfers, nothing to do.
+    if outputs.is_empty() {
+        return Ok(Response::new());
+    }
+
+    // Each transaction can have a maximum number of outputs.
+    let mut events = vec![];
+    let mut iter = outputs.into_iter();
+
+    // Keep creating transactions until there are no more outputs left.
+    while iter.len() > 0 {
+        let mut tx_output = BTreeMap::new();
+        let mut withdraw_amount = Uint128::ZERO;
+
+        for (k, v) in iter.by_ref() {
+            tx_output.insert(k, v);
+            withdraw_amount += v;
+
+            // Check if the maximum number of outputs is reached for this tx.
+            if tx_output.len() == cfg.max_output_per_tx {
```

Comment:
Not sure if this makes sense since there is no maximum number of outputs in Bitcoin. Rather there is weight limits in Consensus (4M WU) and standard relay policy (400k WU). Could update the logic to limit the transactions based on size instead?